protos/flyteidl/admin/execution.proto

```diff
@@ -320,3 +322,25 @@ message WorkflowExecutionGetDataResponse {
     // Full_outputs will only be populated if they are under a configured size threshold.
     core.LiteralMap full_outputs = 4;
 }
+
+message ExecutionUpdateRequest {
+    // Identifier of the execution to update
+    core.WorkflowExecutionIdentifier id = 1;
+
+    // Status object to set as the new value
+    ExecutionStatus status = 2;
+}
+
+message ExecutionStatus {
+    // The state of the execution is used to control its visibility in the UI/CLI.
+    enum ExecutionState {
+        // By default, all executions are considered active.
+        EXECUTION_ACTIVE = 0;
+
+        // Archived executions are no longer visible in the UI.
+        EXECUTION_ARCHIVED = 1;
+    }
+    ExecutionState state = 1;
+}
```

Comment:
Should we store occured_at here as well?

Reply:
Instead of accepting that from the API request which is basically going to be users timestamp of when they execute flytectl update execution <execid> --archive or whichever other form to call the api.

Do we instead want the time stamp to be generated by the system when the request is receieved and store it in the model as StateUpdatedAt timestamp

I have seen in some other PR's where we are doing this system generated timestamps instead of accepting from request.

Having a consistent way of doing this would be preferable in my opinion for such audit related attributes
Let me know what you think

Reply:
True... I'm not suggesting to send this in the request. I think it should be generated by the server but I thought we use the same proto for storage?

Reply:
@EngHabu can you elaborate further on which proto you are talking about.  And also if we are not accepting it in the request then do we need any api changes here . We can handle those directly in consumer flyteadmin right with just the model changes

Reply:
This `message ExecutionStatus {` do we store this in the DB raw? do we know when was it archived? just looking at updated_at seem error prune... 
@katrogan

Reply:
Also my assumption is that this StateUpdatedAt is not exposed in the UI or through CLI. If we do want to show this state then we would need the API change.

Reply:
we shouldn't depend on updated_at because if someone archives an in-progress execution, its termination event will re-set updated_at. @EngHabu just curious, why do we care to store the state change occurred time? could we use the admin audit logs to capture who updated the state and when?

Reply:
@pmahindrakar-oss if we go with your StateUpdatedAt approach won't we need a new protobuf field to expose that, which _will_ require some idl hange?

Reply:
I was thinking it was just for reporting purposes which read directly from the DB but i see if we want queries on this field from UI /ctl then we need to expose it in the API. I have added the field.

Reply:
hey @pmahindrakar-oss we were thinking it might make sense to capture the user in addition to the timestamp when archiving a workflow, like the [principal](https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/admin/execution.proto#L209).

However, we shouldn't let necessarily let users actually specify the principal or the timestamp when issuing the archive request. Rather the user/principal should come from the authenticated context like [here](https://github.com/flyteorg/flyteadmin/blob/master/pkg/manager/impl/execution_manager.go#L750)

So perhaps we could have two separate messages, one that only contains the execution state and that is included in the body of the UpdateExecutionRequest message. Then we would add a separate message containing
-ExecutionState
-OccurredAt
-Principal

which is embedded in the closure and that gets returned for Get/List execution requests

Reply:
I have updated the IDL , please have a look